### ISSUE TITLE
Unable to open in macOS m2 arm64

### ISSUE THREAD

**Issue by superruzafa** [nilsschmidt1337/ldparteditor#965]
**Describe the bug**
Unable to open in macOS m2 arm64

**To Reproduce**
1. Open LDPartEditor in using any of the following methods:
  - Double click on the application in the Applications folder
  - Launch the `ldparteditor.sh` script from terminal
  - `cd /Applications/LDPartEditor.app/Contents/MacOS` and launch `./LDPartEditor`
2. A message saying "LDPardEditor is damaged and can’t be opened." is shown

Additionally if you launch the following command nothing happens:

```
$ cd /Applications/LDPartEditor.app/Contents/app
$ java --version
openjdk 17.0.2 2022-01-18
OpenJDK Runtime Environment (build 17.0.2+8-86)
OpenJDK 64-Bit Server VM (build 17.0.2+8-86, mixed mode, sharing)

$ java -jar LDPartEditor.jar

$ echo $?
1
```

**Expected behavior**
The application is launched

**Desktop (please complete the following information):**
 - OS: macOS Monterey 12.6.5 - Apple arm64 M2
 - Version 1.8.67

**Additional context**
The first time the app is launch [Rosetta](https://en.wikipedia.org/wiki/Rosetta_(software)) installation was prompted, then I installed it.

**Comment by superruzafa** [comment]
The same if I build the app from scratch:

```
$ ant -noinput -buildfile build-macos.xml
Buildfile: /Users/superruzafa/code/ldparteditor/build-macos.xml

create_run_jar:
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/dependencies
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/dependencies/macosx
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/target
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/bin
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/bin/macosx
...

junit:
    [mkdir] Created dir: /Users/superruzafa/code/ldparteditor/report
    [javac] Compiling 3 source files to /Users/superruzafa/code/ldparteditor/target
    [junit] Running org.nschmidt.ldparteditor.Vector3dTest
    [junit] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5,06 sec

BUILD SUCCESSFUL
Total time: 2 minutes 26 seconds
```

**Comment by superruzafa** [comment]
After some research it seems it happens because of the x86_64 version of libswt-cocoa library.

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons:
        no swt-cocoa-4948r9 in java.library.path: /Users/superruzafa/Downloads/swt-4.27-cocoa-macosx-aarch64
        no swt-cocoa in java.library.path: /Users/superruzafa/Downloads/swt-4.27-cocoa-macosx-aarch64
        no swt in java.library.path: /Users/superruzafa/Downloads/swt-4.27-cocoa-macosx-aarch64
        Can't load library: /Users/superruzafa/.swt/lib/macosx/aarch64/libswt-cocoa-4948r9.jnilib
        Can't load library: /Users/superruzafa/.swt/lib/macosx/aarch64/libswt-cocoa.jnilib
        Can't load library: /Users/superruzafa/.swt/lib/macosx/aarch64/libswt.jnilib
        Can't load library: /Users/superruzafa/.swt/lib/macosx/aarch64/libswt-cocoa-4948r9.jnilib

        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:348)
        at org.eclipse.swt.internal.Library.loadLibrary(Library.java:257)
        at org.eclipse.swt.internal.C.<clinit>(C.java:19)
        at org.eclipse.swt.internal.cocoa.NSMutableArray.arrayWithCapacity(NSMutableArray.java:39)
        at org.eclipse.swt.widgets.Canvas.<clinit>(Canvas.java:55)
        at org.eclipse.jface.window.Window.createShell(Window.java:487)
        at org.eclipse.jface.window.Window.create(Window.java:430)
        at org.nschmidt.ldparteditor.splash.SplashScreen.run(SplashScreen.java:152)
        at org.nschmidt.ldparteditor.main.LDPartEditor.main(LDPartEditor.java:49)
```

I replace it by the aarch64 version and now the error I get is

```
java -jar ./bin/macosx/LDPartEditor.jar

***WARNING: Display must be created on main thread due to Cocoa restrictions. Use vmarg -XstartOnFirstThread
org.eclipse.swt.SWTException: Invalid thread access
```

**Comment by nilsschmidt1337** [comment]
@superruzafa , will it work with
`java -XstartOnFirstThread -jar ./bin/macosx/LDPartEditor.jar` ?

I will try to create an alternative "macOS arm64" in the future to support the M2 platform, too.

**Comment by superruzafa** [comment]
@nilsschmidt1337 by passing that flag a dialog is shown to enter user details, parts path, etc... then when I click OK the app crashes

```
[LDPartEditor RELEASE ] Started on Tue May 16 23:13:48 CEST 2023
[LWJGL] Failed to load a library. Possible solutions:
        a) Add the directory that contains the shared library to -Djava.library.path or -Dorg.lwjgl.librarypath.
        b) Add the JAR that contains the shared library to the classpath.
[LWJGL] Enable debug mode with -Dorg.lwjgl.util.Debug=true for better diagnostics.
[LWJGL] Enable the SharedLibraryLoader debug mode with -Dorg.lwjgl.util.DebugLoader=true for better diagnostics.
Exception in thread "main" java.lang.UnsatisfiedLinkError: Failed to locate library: liblwjgl.dylib
        at org.lwjgl.system.Library.loadSystem(Library.java:162)
        at org.lwjgl.system.Library.loadSystem(Library.java:62)
        at org.lwjgl.system.Library.<clinit>(Library.java:50)
        at org.lwjgl.opengl.GL.<clinit>(GL.java:78)
        at org.nschmidt.ldparteditor.splash.SplashScreen.run(SplashScreen.java:184)
        at org.nschmidt.ldparteditor.main.LDPartEditor.main(LDPartEditor.java:49)
```

**Comment by nilsschmidt1337** [comment]
This issue depends on https://github.com/github/roadmap/issues/528.

**Comment by nilsschmidt1337** [comment]
@superruzafa , now it is possible to compile it for arm64 with `ant -noinput -buildfile build-macos-arm.xml`.

**Comment by nilsschmidt1337** [comment]
Closed with commit [d621400](https://github.com/nilsschmidt1337/ldparteditor/commit/d621400e60f5a99c69e3cc613b14e0e7d81dc47f).
